### DISCUSSION TITLE
AC Add-On

### DISCUSSION THREAD

**Discussion by JBENEITEZ** [ENZYME-APD/archicad-api-dev#45] [Ideas]
Hi @kovacsv! This is the list we were discussing earlier in Discord.
https://repeated-second-c61.notion.site/TAPIR-WISH-LIST-42db480a1be8435489ef7048e184e1f3

So we can start the discussion here

**Comment by kovacsv** [comment]
Thanks @JBENEITEZ. So here are some questions regarding the Add-On:
- What will be the name of the Add-On? Tapir Additional JSON Commands?
- Is it ok if it goes under my name at Graphisoft (developer account)?
- Where to put the source? archicad-api/ArchicadAddOn?

**Reply by JBENEITEZ** [reply]
@kovacsv here my opinion others @leceta @thekaushikls @grzegorzwilk please comment too! 

- What will be the name of the Add-On? Tapir Additional Archicad Commands?
_in my opinion that is a good and straightforward name_

- Is it ok if it goes under my name at Graphisoft (developer account)?
_For me, there's no issue with that! It seems the most efficient way to move forward _

- Where to put the source? archicad-api/ArchicadAddOn?
_Regarding this, @thekaushikls was reorganising the repository, maybe we should ask him as well_

**Reply by thekaushikls** [reply]
Hi, those are some good points. Here is what I think.

1. Having the same / similar name makes it easier to promote / recognize.

2. All good.

3. I'd go with a separate repo.
    1. It's easier to maintain.
    2. Easy for contributors to start working on it.
    3. This would make it easy for contributors. One just needs to fork the C++ repo if they want to work on the archicad side.
    4. The repo size might be smaller, and more distributed.
    5. Future extension is simple - just create another repo.
    
    Example:
    1. ```tapir_archicad``` - Extended Archicad JSON API
    2. ```tapir_py``` - Python library that uses the JSON API
    3. ```tapir_gh``` - Grasshopper plugins that use the JSON API
    5. ```tapir.NET``` - (probably in the future) Access the JSON API from any .NET environment.
---
*Alternatively, We can have a **mono-repo** structure. Since the python package, and grasshopper plugins will depend heavily on the archicad package, I think it will not be a big issue.*

What do you think @kovacsv and @leceta ?
